fs/dynchunksize/dynchunksize.go

```diff
@@ -0,0 +1,37 @@
+// Dynamically calculate suitable chunk size for large uploads
+package dynchunksize
+
+import (
+	"github.com/rclone/rclone/fs"
+)
+
+const oneMiB = int64(1024 * 1024)
+
+/*
+ Compute the minimum chunk size needed to fit within the given maximum number of parts, rounded up to the nearest MiB
+
+ For most backends, (chunk_size) * (concurrent_upload_routines) memory will be required so we want to use the smallest
+ possible chunk size that's going to allow the upload to proceed. Rounding up to the nearest MiB on the assumption
+ that some backends may only allow integer type parameters when specifying the chunk size.
+
+ Returns the default chunk size if it is sufficiently large enough to support the given file size otherwise returns the
+ smallest chunk size necessary to allow the upload to proceed.
+*/
+func ComputeChunkSize(fileSize fs.SizeSuffix, maxParts int, defaultChunkSize fs.SizeSuffix) fs.SizeSuffix {
+	requiredChunks := fileSize / defaultChunkSize
+	if requiredChunks < fs.SizeSuffix(maxParts) || (requiredChunks == fs.SizeSuffix(maxParts) && fileSize%defaultChunkSize == 0) {
+		fs.Debugf(nil, "size: %v, parts: %v, default: %v; returned default value", fileSize, maxParts, defaultChunkSize)
```

Comment:
We should probably comment out this debug otherwise it will appear for every chunked upload.